modules/nf-core/mafft/main.nf

```diff
@@ -4,32 +4,35 @@ process MAFFT {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/mafft:7.520--h031d066_3':
-        'biocontainers/mafft:7.520--h031d066_3' }"
+        'https://depot.galaxyproject.org/singularity/mulled-v2-12eba4a074f913c639117640936668f5a6a01da6:425707898cf4f85051b77848be253b88f1d2298a-0':
+        'biocontainers/mulled-v2-12eba4a074f913c639117640936668f5a6a01da6:425707898cf4f85051b77848be253b88f1d2298a-0' }"
 
     input:
-    tuple val(meta),  path(fasta)
+    tuple val(meta) , path(fasta)
     tuple val(meta2), path(add)
     tuple val(meta3), path(addfragments)
     tuple val(meta4), path(addfull)
     tuple val(meta5), path(addprofile)
     tuple val(meta6), path(addlong)
+    val(compress)
 
     output:
-    tuple val(meta), path("*.fas"), emit: fas
-    path "versions.yml"           , emit: versions
+    tuple val(meta), path("*.fas{.gz,}"), emit: fas
+    path "versions.yml"                 , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
     def args         = task.ext.args   ?: ''
     def prefix       = task.ext.prefix ?: "${meta.id}"
-    def add          = add             ? "--add ${add}"                   : ''
-    def addfragments = addfragments    ? "--addfragments ${addfragments}" : ''
-    def addfull      = addfull         ? "--addfull ${addfull}"           : ''
-    def addprofile   = addprofile      ? "--addprofile ${addprofile}"     : ''
-    def addlong      = addlong         ? "--addlong ${addlong}"           : ''
+    def add          = add             ? "--add <(unpigz -cdf ${add})"                   : ''
+    def addfragments = addfragments    ? "--addfragments <(unpigz -cdf ${addfragments})" : ''
+    def addfull      = addfull         ? "--addfull <(unpigz -cdf ${addfull})"           : ''
+    def addprofile   = addprofile      ? "--addprofile <(unpigz -cdf ${addprofile})"     : ''
+    def addlong      = addlong         ? "--addlong <(unpigz -cdf ${addlong})"           : ''
```

Comment:
Ah sorry, missed this on Friday.
I used pigz here to maintain consistency with the output file writing, and as it's already a dependency for processing the output. If gzip is replaced by pigz in the base container, this might become relevant as an implementation using gzip would then require an additional dependency.
Decompression in pigz is by default not parallelized (except for separating the threads for reading/writing), so I wouldn't expect this to have any significant effect on performance.